alternatives.c

```diff
@@ -486,11 +486,13 @@ static int readConfig(struct alternativeSet *set, const char *title,
         }
 
         set->alts = realloc(set->alts, (set->numAlts + 1) * sizeof(*set->alts));
-        set->alts[set->numAlts++] = newAlt;
+        set->alts[set->numAlts] = newAlt;
 
-        if (newAlt.priority > set->alts[set->best].priority)
+        if (set->best == -1 || newAlt.priority > set->alts[set->best].priority)
             set->best = set->numAlts;
```

Comment:
Ah, I see... I missed that off-by-one. Fixed in the 2nd commit, where it really belongs.